docs/articles/documentation/using-testcafe/configuration-file.md

```diff
@@ -0,0 +1,764 @@
+See [Smart Assertion Query Mechanism](../test-api/assertions/README.md#smart-assertion-query-mechanism).
+
+```json
+{
+    "assertionTimeout": 1000
+}
+```
+
+**Default value**: `3000`
+
+*CLI*: [--assertion-timeout](command-line-interface.md#--assertion-timeout-ms)  
+*API*: [runner.run({ assertionTimeout })](programming-interface/runner.md#run)
+
+## pageLoadTimeout
+
+Specifies the time (in milliseconds) passed after the `DOMContentLoaded` event, within which TestCafe waits for the `window.load` event to fire.
+
+After the timeout passes or the `window.load` event is raised (whichever happens first), TestCafe starts the test.
+
+```json
+{
+    "pageLoadTimeout": 1000
+}
+```
+
+**Default value**: `3000`
+
+See the command line [--page-load-timeout](command-line-interface.md#--page-load-timeout-ms) parameter for details.
+
+*CLI*: [--page-load-timeout](command-line-interface.md#--page-load-timeout-ms)  
+*API*: [runner.run({ pageLoadTimeout })](programming-interface/runner.md#run)
+
+## speed
+
+Specifies the test execution speed.
+
+Tests are run at the maximum speed by default. You can use this option
+to slow the test down.
+
+Provide a number between `1` (the fastest) and `0.01` (the slowest).
+
+```json
+{
+    "speed": 0.1
+}
+```
+
+**Default value**: `1`
+
+If the speed is also specified for an [individual action](../test-api/actions/action-options.md#basic-action-options), the action's speed setting overrides the test speed.
+
+*CLI*: [--speed](command-line-interface.md#--speed-factor)  
+*API*: [runner.run({ speed })](programming-interface/runner.md#run)
+
+## port1, port2
+
+Specifies custom port numbers TestCafe uses to perform testing. The number range is [0-65535].
+
+```json
+{
+    "port1": 12345,
+    "port2": 54321
+}
+```
+
+TestCafe automatically selects ports if ports are not specified.
+
+*CLI*: [--ports](command-line-interface.md#--ports-port1port2)  
+*API*: [createTestCafe](programming-interface/createtestcafe.md)
+
+## hostname
+
+Specifies your computer's hostname. It is used when you run tests in remote browsers.
+
+```json
+{
+    "hostname": "host.mycorp.com"
+}
+```
+
+If the hostname is not specified, TestCafe uses the operating system's hostname or the current machine's network IP address.
+
+*CLI*: [--hostname](command-line-interface.md#--hostname-name)  
+*API*: [createTestCafe](programming-interface/createtestcafe.md)
+
+## proxy
+
+Specifies the proxy server used in your local network to access the Internet.
+
+```json
+{
+    "proxy": "proxy.corp.mycompany.com"
+}
+```
+
+```json
+{
+    "proxy": "172.0.10.10:8080"
+}
+```
+
+You can also specify authentication credentials with the proxy host.
+
+```json
+{
+    "proxy": "username:password@proxy.mycorp.com"
+}
+```
+
+*CLI*: [--proxy](command-line-interface.md#--proxy-host)  
+*API*: [runner.useProxy](programming-interface/runner.md#useproxy)
+
+## proxyBypass
+
+Specifies the resources accessed bypassing the proxy server.
```

Comment:
accessed bypassing?

Reply:
(the same reply as in the previous case)